spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/config/HttpClientFactory.java

```diff
@@ -69,12 +49,15 @@ public class HttpClientFactory extends AbstractFactoryBean<HttpClient> {
 
 	protected final ServerProperties serverProperties;
 
+	protected final HttpClientSslConfigurer sslConfigurer;
+
 	protected final List<HttpClientCustomizer> customizers;
 
 	public HttpClientFactory(HttpClientProperties properties, ServerProperties serverProperties,
-			List<HttpClientCustomizer> customizers) {
+			HttpClientSslConfigurer sslConfigurer, List<HttpClientCustomizer> customizers) {
```

Comment:
Since this is a public api, the old constructor needs to remain and be marked deprecated passing in a default for `HttpClientSslConfigurer`. It can then be removed from main.

Reply:
I understand we are talking about something like

```java
	public HttpClientFactory(HttpClientProperties properties, ServerProperties serverProperties,
			List<HttpClientCustomizer> customizers) {
		this.properties = properties;
		this.serverProperties = serverProperties;
		this.sslConfigurer = new HttpClientSslConfigurer(properties.getSsl(), serverProperties);
		this.customizers = customizers;
	}
```

But it that case, wouldn't it make sense to remove the new constructor? If users want to customize the ssl configuration, they can override the restored `HttpClient configureSsl(HttpClient httpClient)`.

Reply:
Or even instantiate the HttpClientSslConfigurer inside the `configureSsl` method as
```
	protected HttpClient configureSsl(HttpClient httpClient) {
		return new HttpClientSslConfigurer(properties.getSsl(), serverProperties)
				.configureSsl(httpClient);
	}
```